pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java

```diff
@@ -60,9 +69,91 @@ public void tearDown() {
+        }
+
+        final DeviceDriverManager deviceDriverManager = new DeviceDriverManager(properties);
+        Service startService = deviceDriverManager.startAsync();
+        try {
+            startService.awaitRunning(Duration.ofSeconds(30));
+            Thread.sleep(25000);
+        } catch (InterruptedException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+        final Connection connection = SQliteDBUtility.createDatabase(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE"));
+        final TransactionStateDB state = new TransactionStateSQLiteImpl(connection, null);
+
+        try {
+            List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
+            Assertions.assertEquals(1, completedFiles.size());
+
+            URI controllerURI = setupUtils.getControllerUri();
+            String scope = properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_SCOPE");
+            String streamName = properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_STREAM");
+
+            StreamManager streamManager = StreamManager.create(controllerURI);
+
+            final String readerGroup = UUID.randomUUID().toString().replace("-", "");
+            final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                    .stream(Stream.of(scope, streamName))
+                    .build();
+            try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
+                readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+            }
+
+            try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope,
+                    ClientConfig.builder().controllerURI(controllerURI).build());
+                 EventStreamReader<String> reader = clientFactory.createReader("reader",
+                         readerGroup,
+                         new UTF8StringSerializer(),
+                         ReaderConfig.builder().build())) {
+                System.out.format("Reading all the events from %s/%s%n", scope, streamName);
+                EventRead<String> eventRead = null;
+                try {
+                    while ((eventRead = reader.readNextEvent(2000)).getEvent() != null) {
+                        String event = eventRead.getEvent();
+                        System.out.format("Read event: %s", event);
+                        Assertions.assertNotNull(event);
+                        Assertions.assertFalse(event.isEmpty());
+                        Assertions.assertEquals("Hello World.", event);
+                    }
+                } catch (ReinitializationRequiredException e) {
+                    //There are certain circumstances where the reader needs to be reinitialized
+                    e.printStackTrace();
+                }
+                System.out.format("No more events from %s/%s%n", scope, streamName);
+            }
+
+            Thread.sleep(50000);
```

Comment:
Is this sleep intentional ? why 50K?

Reply:
reduced sleep interval to 5 seconds from 50 seconds.